users/tests/test_pipeline.py

```diff
@@ -103,3 +106,35 @@ def test_association_by_keycloak_uuid(login_backend, existing_backend, user, exp
         assert pipeline_data == {"user": user}
     else:
         assert pipeline_data is None
+
+
+@pytest.mark.parametrize(
+    "backend_name,expected",
+    (
+        ("heltunnistussuomifi", True),
+        ("helsinki_tunnus", True),
+        ("helsinki_adfs", False),
+    ),
```

Comment:
If I see something called `expected` in unit tests, I expect it to be the expected value. Instead, this is a flag for changing the expected value later in the test. Just change this to the actual expected values.

If you're worried about `None == None` (which should be fine in this case IMO), just set a default value in `session_get`, e.g. `session_get("kc_action_status", "NOT SET")`. Or at least I imagine it should have a default option. Then you can definitely just assert for `strategy.blabla == expected`.

Reply:
Refactored. :+1: